ledger/src/blockstore.rs

```diff
@@ -2915,46 +2918,70 @@ impl Blockstore {
         &self,
         bank: &Bank,
         starting_slot: Slot,
         ending_slot: Slot,
-    ) -> DashSet<Pubkey> {
+    ) -> (DashSet<Pubkey>, bool) {
         let result = DashSet::new();
+        let lookup_tables = DashSet::new();
+        let possible_cpi_alt_extend = AtomicBool::new(false);
 
         (starting_slot..=ending_slot)
             .into_par_iter()
             .for_each(|slot| {
                 if let Ok(entries) = self.get_slot_entries(slot, 0) {
                     entries.into_par_iter().for_each(|entry| {
                         entry.transactions.into_iter().for_each(|tx| {
+                            for account in tx.message.static_account_keys() {
+                                result.insert(*account);
+                            }
+
                             if let Some(lookups) = tx.message.address_table_lookups() {
                                 lookups.iter().for_each(|lookup| {
-                                    result.insert(lookup.account_key);
+                                    lookup_tables.insert(lookup.account_key);
                                 });
                             }
-                            // howdy, anybody who reached here from the panic messsage!
-                            // the .unwrap() below could indicate there was an odd error or there
-                            // could simply be a tx with a new ALT, which is just created/updated
-                            // in this range. too bad... this edge case isn't currently supported.
-                            // see: https://github.com/solana-labs/solana/issues/30165
-                            // for casual use, please choose different slot range.
-                            let sanitized_tx = bank.fully_verify_transaction(tx).unwrap();
-                            sanitized_tx
-                                .message()
-                                .account_keys()
-                                .iter()
-                                .for_each(|&pubkey| {
-                                    result.insert(pubkey);
-                                });
+
+                            let tx = SanitizedVersionedTransaction::try_from(tx)
+                                .expect("transaction failed to sanitize");
+                            let alt_scan_result =
+                                transaction_address_lookup_table_scanner::scan_transaction(&tx);
+                                match alt_scan_result {
+                                    transaction_address_lookup_table_scanner::ScanResult::NotFound => {},
+                                    transaction_address_lookup_table_scanner::ScanResult::NativeUsed(keys) => {
+                                        keys.into_iter().for_each(|key| { result.insert(key); });
+                                    }
+                                    transaction_address_lookup_table_scanner::ScanResult::NonNativeUsed(keys) => {
```

Comment:
nit: how about `use transaction_address_lookup_table_scanner::{...}`-ing?

Reply:
Yeah didn't do that originally because the `ScanResult` name was terrible and too generic. Now that that is going to change I'll just `use` here